app/values/file_info.rb

```diff
@@ -0,0 +1,24 @@
+class FileInfo
+  attr_reader :file_info
+  delegate :[], :has_key?, to: :file_info
+  def initialize(file_info)
+    @file_info = file_info
+  end
+
+  def to_h
+    file_info
+  end
+
+  def file_name
+    file_info["file_name"]
+  end
+
+  def file_path
+    FilePath.new(file_info["url"])
+  end
+
+  def mime_type
```

Comment:
As far as I can tell, `file_info` doesn't come with `"mime_type"`. I could add it with the idea that maybe some services do add it? It looks like maybe dropbox does?